plugins/database/mongodb/connection_producer.go

```diff
@@ -29,14 +26,23 @@ type mongoDBConnectionProducer struct {
 	Username      string `json:"username" structs:"username" mapstructure:"username"`
 	Password      string `json:"password" structs:"password" mapstructure:"password"`
 
-	Initialized bool
-	RawConfig   map[string]interface{}
-	Type        string
-	session     *mgo.Session
-	safe        *mgo.Safe
+	Initialized   bool
+	RawConfig     map[string]interface{}
+	Type          string
+	clientOptions *options.ClientOptions
+	client        *mongo.Client
 	sync.Mutex
 }
 
+// WriteConcern defines the write concern options
+type WriteConcern struct {
```

Comment:
I don't think we need to export this since we're only using it to unmarshal.

Reply:
Thanks. Makes sense.